.buildkite/pipeline_mflowgen.yml

```diff
@@ -22,10 +22,90 @@ steps:
   continue_on_failure: true
 
 
-
-
+##############################################################################
+# FULL CHIP RUNS
+# 
+# To build the full chip through LVS, starting from scratch:
+#   (Note if you do it all at one go like this, it doesn't save anything
+#   til the very end, and then only if nothing failed along the way! :(
+# 
+# - label: 'full_chip 36h'
+#   agents: { jobsize: "hours" }
+#   commands:
+#   - 'mflowgen/test/test_module.sh full_chip
+#        --debug
+#        --update_cache /sim/buildkite-agent/gold.$$BUILDKITE_BUILD_NUMBER
+#        --steps lvs'
+# 
+# To build the full chip through LVS, starting from scratch, with discrete checkpoints
+# - label: 'full_chip 36h'
+#   agents: { jobsize: "hours" }
+#   commands:
+#   - 'mflowgen/test/test_module.sh full_chip
+#        --steps tile_array,dragonphy,global_controller,init,place,routelvs'
```

Comment:
what is routelvs?

Reply:
Also, what do you mean by discrete checkpoints here? Does this cause all of the targets you list to be pushed to the mflowgen stash? Simply running LVS would run all of these targets, assuming routelvs is supposed to be route,lvs

Reply:
> 
> 
> what is routelvs?

oops, good catch, that should be "route,lvs"

Reply:
> ... Also, what do you mean by discrete checkpoints here? ...

"Discrete checkpoints" in this case means the following: if you say "test_module.sh full_chip --steps lvs" then buildkite runs "make lvs", 36 hours, all one go. If you say "test_module full_chip --steps tile_array,place,route,lvs" then buildkite runs, sequentially as separate commands
```
make tile_array
make cadence-innovus-place
make cadence-innovus-route
make mentor-calibre-lvs
```

...does that make sense?

Reply:
Yes that makes sense, but what is the point of doing it? Intermediate results are always saved regardless.

Reply:
> Yes that makes sense, but what is the point of doing it? Intermediate results are always saved regardless.

If you're saving results using e.g. `--update_cache /sim/tmp/savedir17`, and you're doing it all in one go e.g. `--steps lvs`, and the test script crashes after cadence-innovus-route, then nothing gets saved.

If you're saving results and you're doing it as multiple discrete steps e.g. `--steps place,route,lvs`, then the cache dir `/sim/tmp/savedir17` gets updated after each step. Now, if the test script crashes after cadence-innovus-route, you have a lot of saved state that doesn't have to be repeated.

This has saved my bacon more than once...

2) If you're *not* saving results in a cache, it's still useful, because each individual step appears as a separate menu item in the buildkite log. Making it easier to find problems in whatever step you're trying to focus on, instead of having to page through one huge "make lvs" log.

Reply:
Ok, fair enough. When you say saved, you mean put in the cache. I mean that the intermediate results exist regardless of how it's run (all at once or in "discrete steps"). Seems like it would be good to have Buildkite automatically cache all of these intermediate results after a failure so that failures are easy to diagnose and continue on from, but that's a larger change for a future PR.